### ISSUE TITLE
Wrong SQL generated when outer and correlated subquery share implicit join path

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#10992]
This doesn't work:

```java
TA a = T_A.as("a");

List<Integer> r1 =
create().select(a.tA().B1)
        .from(a)
        .where(exists(
            select(a.tA().B1)
            .from(a)
        ))
        .fetch(a.tA().B1);
```

It produces a wrong SQL query:

```sql
select alias_53273784.B1
from T_A a -- Missing join here
where exists (
  select alias_53273784.B1
  from (
    T_A a
      left outer join T_A alias_53273784
        on a.PARENT_A = alias_53273784.A
    )
)
```

The error being:

> Column "ALIAS_53273784.B1" not found;

This works:

```java
TA a = T_A.as("a");

List<Integer> r1 =
create().select(a.tA().B1)
        .from(a)
        .where(exists(
            select(T_A.tA().B1) // Don't use aliased table a here
            .from(T_A)
        ))
        .fetch(a.tA().B1);
```

Producing the correct implicit join paths (two separate paths with different aliases):

```sql
select alias_53273784.B1
from (
  T_A a
    left outer join T_A alias_53273784
      on a.PARENT_A = alias_53273784.A
  )
where exists (
  select alias_43932461.B1
  from (
    T_A
      left outer join T_A alias_43932461
        on T_A.PARENT_A = alias_43932461.A
    )
)
```

This issue causes https://github.com/jOOQ/jOOQ/issues/10940, a regression because of a query that now uses this syntax.

**Comment by lukaseder** [comment]
This test also reproduces it:

```java
List<Integer> r2 =
create().select(a.tA().B1)
        .from(a)
        .where(a.tA().B1.in(
            select(a.tA().B1)
            .from(a)
        ))
        .orderBy(a.A)
        .fetch(a.tA().B1);

assertEquals(asList(1, 1, 2, 2), r2);
```

Producing a similar invalid query:

```sql
select alias_53273784.B1
from T_A a
where alias_53273784.B1 in (
  select alias_53273784.B1
  from (
    T_A a
      left outer join T_A alias_53273784
        on a.PARENT_A = alias_53273784.A
    )
)
order by a.A
```

**Comment by lukaseder** [comment]
The bug is essentially that we currently cannot hide an identifier from an outer scope in an inner scope. The nested declaration of `T_A as a` should prevent access to the table `a` from the outer scope.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.4 (#11005)
